WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift

```diff
@@ -253,7 +253,15 @@ class MediaLibraryViewController: WPMediaPickerViewController {
     }
 
     private func showOptionsMenu() {
-        let pickingContext = MediaPickingContext(origin: self, view: view, barButtonItem: navigationItem.rightBarButtonItem, blog: blog)
+
+        let pickingContext: MediaPickingContext = {
+            if pickerDataSource.totalAssetCount > 0 {
+                return MediaPickingContext(origin: self, view: view, barButtonItem: navigationItem.rightBarButtonItem, blog: blog)
+            } else {
+                return MediaPickingContext(origin: self, view: noResultsView.actionButton, blog: blog)
+            }
+        }()
+
```

Comment:
I'd like to suggest two slightly more concise alternatives to the preceding block.

Option A:
```
let barButtonItem: UIBarButtonItem?
if pickerDataSource.totalAssetCount > 0 {
    barButtonItem = navigationItem.rightBarButtonItem
} else {
    barButtonItem = nil
}
let pickingContext = MediaPickingContext(origin: self, view: view, barButtonItem: barButtonItem, blog: blog)
```

Option B:
```
let pickingContext: MediaPickingContext
if pickerDataSource.totalAssetCount > 0 {
    pickingContext = MediaPickingContext(origin: self, view: view, barButtonItem: navigationItem.rightBarButtonItem, blog: blog)
} else {
    pickingContext = MediaPickingContext(origin: self, view: noResultsView.actionButton, blog: blog)
}
```

Do you think either of those merit consideration?